javascript/javascript2/week1/cardio.js

```diff
@@ -0,0 +1,66 @@
+// Find the shortest word
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+
+function getShortestWord(arr) {
+  let shortestWord = arr[0]; // we assume that the first word in the array is the shortes one
+  for (let i = 0; i < arr.length; i++) {
+    if (shortestWord.length > arr[i].length) {
+      shortestWord = arr[i];
+    } else {
+      continue;
+    }
+  }
+  return shortestWord;
+}
+
+//Test
+console.log(getShortestWord(danishWords));
+
+//Find and count the Danish letters
+const danishString = "Jeg har en blå bil";
+const danishString2 = "Blå grød med røde bær";
+
+function findDanishLetters(str) {
+  const danishLetters = {
+    total: 0,
+    å: 0,
+    æ: 0,
+    ø: 0,
+  }; // creating an initial object with unique danish letters as keys
+
+  for (
+    let i = 0;
+    i < str.length;
+    i++ // iterating through a string to identify if a letter is danish or not
+  ) {
+    switch (str[i]) {
+      case "æ":
+        danishLetters.æ += 1;
+        danishLetters.total += 1;
+        break;
+      case "å":
+        danishLetters.å += 1;
+        danishLetters.total += 1;
+        break;
+      case "ø":
+        danishLetters.ø += 1;
+        danishLetters.total += 1;
+        break;
+      default:
+        break;
+    }
+  }
+
+  Object.keys(danishLetters).forEach(
+    (k) => danishLetters[k] == 0 && delete danishLetters[k]
+  ); // deleting those keys with zero values
```

Comment:
Great stuff! A perfectly fine solution. One thing to think about is: what would happen if we're matching against a larger set of characters, say the entire alphabet or something? Should we have a switch statement with 28 cases?  
 
There's a more concise way of doing this, which amounts to checking if the key exists in the object and responding accordingly. So you could write something like this: 
 
```javascript
function countDanishLetters(danishString){
    const danishLetters = ['æ', 'ø', 'å'];
    const tally = {};

    let total = 0;
    for(char of danishString) {
        if (danishLetters.includes(char)) {
            tally[char] = !tally[char] ? 1 : tally[char] + 1;
            total += 1;
        }
    }

    tally['total'] = total;
    return tally;
}
```
Here we're checking if a character is a key in the object with `!tally[char]`, which will return `true` if the character is missing. In that case we initialize it to 1, otherwise we add one to what is already there. 
 
You don't need to change anything, but it might be good to know in the future :)

Reply:
Thanks a lot for the hint! Gonna use it in future :)